include/wlr/types/wlr_client_buffer.h

```diff
@@ -13,9 +13,25 @@
 #include <wayland-server-core.h>
 #include <wlr/types/wlr_buffer.h>
 
+struct wlr_renderer;
+struct wlr_client_buffer;
+
+struct wlr_client_buffer_impl {
+	bool (*is_instance)(struct wl_resource *resource);
+	bool (*get_buffer_size)(struct wl_resource *resource,
+			struct wlr_renderer *renderer, int *width, int *height);
+	bool (*initialize)(struct wlr_client_buffer *,
+			struct wlr_renderer *renderer);
```

Comment:
`initialize` and `terminate` are pretty weird here. The wording should at least be aligned with wlroots' (`init`/`finish`, see CONTRIBUTING).

Do client buffer impls need to store additional fields? Maybe it'd be better to have a function that looks like this?

```c
struct wlr_client_buffer *(*create)(struct wl_resource *resource, struct wlr_renderer *renderer);
```

Reply:
I don't have a use case for custom buffer storing additional. Maybe we could expose both `init` and `create`? IMO there's a lot of stuff that the user could miss by creating the buffer themselves.

Reply:
Seems like create/destroy makes the most sense, will do the changes.